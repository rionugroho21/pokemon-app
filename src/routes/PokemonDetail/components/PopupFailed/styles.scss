.popup-failed {
  width: 100%;
  text-align: center;
  display: block;
}