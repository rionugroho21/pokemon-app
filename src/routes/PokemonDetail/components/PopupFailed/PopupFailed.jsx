import React from 'react';
import Modal from 'react-bootstrap/Modal';

import './styles.scss';

export const PopupCatch = props => {
  const {
    open,
    message,
    handleModal
  } = props;

  return (
    <Modal
      show={open}
      onHide={() => handleModal(false)}
      size="lg"
      aria-labelledby="contained-modal-title-vcenter"
      centered
    >
      <Modal.Header>
        <Modal.Title id="contained-modal-title-vcenter" className="popup-failed">
          {message}
        </Modal.Title>
      </Modal.Header>
    </Modal>
  );
};

export default PopupCatch;