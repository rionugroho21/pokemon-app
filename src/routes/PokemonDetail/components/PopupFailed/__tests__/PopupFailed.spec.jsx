import React from 'react';
import { render, cleanup } from '@testing-library/react';

import PopupFailed from '../PopupFailed';

const props = {
  open: false,
  message: 'Failed to catch the Pokemon',
  handleModal: jest.fn()
};

describe('PopupFailed Page', () => {
  afterEach(cleanup);

  it('should render page correctly', () => {
    const { asFragment } = render(<PopupFailed {...props} />);
    expect(asFragment()).toMatchSnapshot();
  });
});