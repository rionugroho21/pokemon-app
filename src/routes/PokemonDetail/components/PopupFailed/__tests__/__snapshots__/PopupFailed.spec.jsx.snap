// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PopupFailed Page should render page correctly 1`] = `<DocumentFragment />`;
